pkgs/tools/security/pinentry/default.nix

```diff
@@ -1,100 +1,18 @@
-{ fetchurl, mkDerivation, fetchpatch, stdenv, lib, pkg-config, autoreconfHook, wrapGAppsHook
-, libgpg-error, libassuan, qtbase, wrapQtAppsHook
-, ncurses, gtk2, gcr
-, withLibsecret ? true, libsecret
-, enabledFlavors ? [ "curses" "tty" "gtk2" "emacs" ]
-  ++ lib.optionals stdenv.isLinux [ "gnome3" ]
-  ++ lib.optionals (!stdenv.isDarwin) [ "qt" ]
+{ callPackage
+, withLibsecret ? true
 }:
 
-assert lib.isList enabledFlavors && enabledFlavors != [];
-
 let
-  pinentryMkDerivation =
-    if (builtins.elem "qt" enabledFlavors)
-      then mkDerivation
-      else stdenv.mkDerivation;
-
-  enableFeaturePinentry = f:
-    let
-      flag = flavorInfo.${f}.flag or null;
-    in
-      lib.optionalString (flag != null)
-        (lib.enableFeature (lib.elem f enabledFlavors) ("pinentry-" + flag));
-
-  flavorInfo = {
-    curses = { bin = "curses"; flag = "curses"; buildInputs = [ ncurses ]; };
-    tty = { bin = "tty"; flag = "tty"; };
-    gtk2 = { bin = "gtk-2"; flag = "gtk2"; buildInputs = [ gtk2 ]; };
-    gnome3 = { bin = "gnome3"; flag = "gnome3"; buildInputs = [ gcr ]; nativeBuildInputs = [ wrapGAppsHook ]; };
-    qt = { bin = "qt"; flag = "qt"; buildInputs = [ qtbase ]; nativeBuildInputs = [ wrapQtAppsHook ]; };
-    emacs = { bin = "emacs"; flag = "emacs"; buildInputs = []; };
-  };
-
+  buildPinentry = pinentryExtraPname: buildFlavors:
+    callPackage ./generic.nix {
```

Comment:
I do not think `override` works as expected. The `override` added by `callPackages` in `all-packages.nix`  will shadow the `override` added by this `callPackage`.

I do not see the point in splitting it into a separate file either.

Reply:
Hmm, this does work though: https://github.com/NixOS/nixpkgs/pull/133542/files#diff-9bb1154556d469c1e3ebb8013675e90103eacdba65f86b80a30335717d6ceb6fR67

I'm gonna test.

Reply:
Right, this does not `pinentry-all.override {gtk2 = gtk2.overrideAttrs {…};}`, though.

Reply:
Thanks, was indeed an issue. I changed that very late and didn't test properly again. Now all in one file again.